### PR TITLE
Travis: Build master with ghc-8.2

### PR DESCRIPTION
**Pull request by alexbiehl**
We were using `ghc-head` for the master branch.

### CODE REVIEW

**Comment by alexbiehl** [comment]
@hvr If this looks good for you I will land this and fix the testsuite failures in subsequent PR (namely https://github.com/haskell/haddock/pull/602).

File: haddock.cabal [hunk 1]
```diff
@@ -12,9 +12,10 @@ bug-reports:          https://github.com/haskell/haddock/issues
 copyright:            (c) Simon Marlow, David Waern
 category:             Documentation
 build-type:           Simple
-cabal-version:        >= 1.10
+cabal-version:        >= 2.0
```
Comment (hvr) [inline haddock.cabal]: We'd need to check whether GHC HEAD can still be bootstrapped with this (GHC's build uses only the  `haddock.cabal` file). Would `buildtools: haddock` work as well?

Comment (hvr) [inline haddock.cabal]: I've tweaked `cabal-version` in GHC HEAD, and ran a `./validate` and it works just fine. So false alarm.

**Comment by alexbiehl** [comment]
Thanks @hvr!

**Pull request merged**